vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+  pillar.width = Inf, # show all columns
+  pillar.min_chars = Inf # to turn off truncation of column titles
+)
+
+myDataCombined$toDataFrame()
+```
+
+
+```{r, echo=FALSE}
+# change these settings only temporarily
+withr::with_options(
+  list(
+    pillar.width = Inf, # show all columns
+    pillar.min_chars = Inf # to turn off truncation of column titles
+  ),
+  code = {
+    print(myDataCombined$toDataFrame())
+  }
+)
+```
+
+As can be seen, this data frame attemtps to provide a one-stop access to all important details about the datasets included in the object, including important meta-data.
+
+`{ospsuite}` also provides a few helper functions to modify the data frame further.
+
+When multiple (observed and/or simulated) datasets are present in a data frame, they are likely to have different units. `convertUnits()` function helps to convert them to a common unit^[Note that if you are using `DataCombined` object for plotting functions, you don't need to this conversion; the plotting functions will take care of this internally.].
+
+```{r}
+convertUnits(
+  myDataCombined,
+  xUnit = ospUnits$Time$s,
+  yUnit = ospUnits$`Concentration [mass]`$`µg/l`
+)
+```
+
+## Further functionalities
+
+Grouping of simulated results with observed data inside `DataCombined` also allows 
+to calculate the error between simulated results and observations using the 
+`calculateResiduals()` function. The function calculates the distance between 
+each observed data point within the group and the simulated results. Simulation 
+results interpolated if no simulated value is available for a certain observed 
+time point.
+
+There are two ways to calculate the distance (residuals) - linear and logarithmic - 
+specified by the `scaling` argument. For linear scaling (`scaling = tlf::Scaling$lin`):
+
+> Residuals are calculated as: Simulation value - Observed value. This means that the residuals are defined by absolute differences.
+
+For logarithmic scaling (`scaling = tlf::Scaling$log`):
+
+> Residuals are calculated as: log(Simulation value) - log(Observed value) = log (Simulation Value / Observed Value). This means that the ratio of values is considered which is independent of the magnitude of the value. But for very small observed values, in particular close to 0 values, this can lead to problems, because log(10E-N) = -N can becomes large.
+
+
+```{r}
+#Linear residuals
+calculateResiduals(myDataCombined, scaling = tlf::Scaling$lin)$residualValues
+
+#Logarithmic residuals
+calculateResiduals(myDataCombined, scaling = tlf::Scaling$log)$residualValues
+```
+
+To quickly calculate the total error of the `DataCombined`, one can sum up the 
+absolute values of the residulas:
+
+```{r}
+#Linear residuals
+totalError <- sum(abs(calculateResiduals(myDataCombined, scaling = tlf::Scaling$lin)$residualValues))
+
+print(totalError)
+```
+
+## Visualizations with `DataCombined`
+
+Plotting the stored data is an important part of model diagnostics and quality control.
```

Comment:
Would remove this sentence. I think we should not state the obvious things

Reply:
Removed.